src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIJsonContext.cs

```diff
@@ -11,6 +12,7 @@ namespace Microsoft.Extensions.AI;
     UseStringEnumConverter = true,
     DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
     WriteIndented = true)]
-[JsonSerializable(typeof(OpenAIChatClient.OpenAIChatToolJson))]
 [JsonSerializable(typeof(OpenAIRealtimeExtensions.ConversationFunctionToolParametersSchema))]
+[JsonSerializable(typeof(OpenAIModelMappers.OpenAIChatToolJson))]
+[JsonSerializable(typeof(IDictionary<string, object?>))]
```

Comment:
Were we just missing this by accident before? If so it's a shame our tests can't catch it. Or maybe this type was included transitively and you're adding it here to be more explicit.

Reply:
It's a new addition -- rather than speculatively using `ToolCallJsonSerializerOptions.GetTypeInfo(typeof(IDictionary<string, object>))` I'm hardcoding a source generated contract for the dictionary. This is because `object` always gets deserialized as `JsonElement` under the hood so it doesn't require user customization.